vue/src/components/UploadForm.vue

```diff
@@ -73,8 +64,9 @@ export default {
       formData.append('fileToUpload', this.file);
       this.status = STATUS_UPLOADING;
       try {
-        await axios.post('/upload', formData);
+        const { data } = await axios.post('/upload', formData);
         this.status = STATUS_SUCCESS;
+        this.$emit('done', data);
       } catch (err) {
         this.errorMessage = err;
         this.status = STATUS_FAILED;
```

Comment:
do we need to $emit anything here to notify app.vue of the failure code? or will it be able to access that directly from UploadForm.status?

Reply:
I don't think the App.vue needs to know about failures, since this UploadForm.vue component resets itself on failure so the user can try another upload. There's nothing for App.vue to act on in case of failure here.